packages/shared-ux/avatar/solution/src/solution_avatar.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import './solution_avatar.scss';
+
+import React from 'react';
+import classNames from 'classnames';
+
+import { DistributiveOmit, EuiAvatar, EuiAvatarProps, IconType } from '@elastic/eui';
+import { EuiIconType } from '@elastic/eui/src/components/icon/icon';
+
+type FilterStartingWith<Set, Needle extends string> = Set extends `${Needle}${infer _X}`
+  ? Set
+  : never;
+
+type ExtractSolutionName<SolutionString extends string> =
+  SolutionString extends `logo${infer Solution}` ? Solution : { error: 'Cannot parse logo' };
+
+type SolutionNameType = ExtractSolutionName<FilterStartingWith<EuiIconType, 'logo'>>;
+
+export type KnownSolutionProps = DistributiveOmit<EuiAvatarProps, 'size' | 'name' | 'iconType'> & {
+  /**
+   * Any EuiAvatar size available, or `xxl` for custom large, brand-focused version
+   */
+  size?: EuiAvatarProps['size'] | 'xxl';
+  name: SolutionNameType;
+};
+
+export type IconTypeProps = DistributiveOmit<EuiAvatarProps, 'size' | 'name' | 'iconType'> & {
+  /**
+   * Any EuiAvatar size available, or `xxl` for custom large, brand-focused version
+   */
+  size?: EuiAvatarProps['size'] | 'xxl';
+  name?: string;
```

Comment:
Don't we want to constraint the names allowed so ``logo${rest.name}`` belows render an existing logo?

Reply:
I just see in the gif in the description that it does. 😄 Interesting...